src/Services/Warehouse/Command/Application/DependencyInjection/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,27 @@
+using Application.Abstractions;
+using Application.Services;
+using Application.UseCases.Commands;
+using Application.UseCases.Events;
+using Contracts.Services.Warehouse;
+using Microsoft.Extensions.DependencyInjection;
+using ShoppingCartDomainEvent = Contracts.Services.ShoppingCart.DomainEvent;
+
+namespace Application.DependencyInjection.Extensions;
+
+public static class ServiceCollectionExtensions
+{
+    public static IServiceCollection AddApplicationServices(this IServiceCollection services)
+        => services
+            .AddScoped<IApplicationService, ApplicationService>();
+
+    public static IServiceCollection AddCommandInteractors(this IServiceCollection services)
+        => services
+            .AddScoped<IInteractor<Command.CreateInventory>, CreateInventoryInteractor>()
+            .AddScoped<IInteractor<Command.DecreaseInventoryAdjust>, DecreaseInventoryAdjustInteractor>()
+            .AddScoped<IInteractor<Command.IncreaseInventoryAdjust>, IncreaseInventoryAdjustInteractor>()
+            .AddScoped<IInteractor<Command.ReceiveInventoryItem>, ReceiveInventoryItemInteractor>();
+
+    public static IServiceCollection AddEventInteractors(this IServiceCollection services)
+        => services
+            .AddScoped<IInteractor<ShoppingCartDomainEvent.CartItemAdded>, CartItemAddedInteractor>();
```

Comment:
Event Interactor must be injected by the interface, based on broadcast

Reply:
done